### PR TITLE
rust: opt 3

### PR DESCRIPTION
**Pull request by chenyan-dfinity**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -18,7 +18,7 @@ members = [
 debug = false
 panic = "abort"
 lto = true
-opt-level = 2
+opt-level = 3
```
Comment (ggreif) [inline Cargo.toml:21]: These numbers are very promising, maybe we should switch to using `-O3` for the RTS too. @luc-blaeser did you ever try this?

Comment (luc-blaeser) [reply Cargo.toml:21]: Not yet. I can try this and measure with the GC benchmark...

Comment (chenyan-dfinity) [reply Cargo.toml:21]: Surprisingly, -O2 is even better than -O3, see https://github.com/dfinity/canister-profiling/pull/62

Comment (chenyan-dfinity) [reply Cargo.toml:21]: Confirmed from https://github.com/dfinity/canister-profiling/pull/66, the default release profile is already using -O3, the gain here purely comes from LTO.

**Pull request closed without merging**